src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1219 @@
+
+            # take what input data we were given and try to use it to
+            # construct polynomials
+            self._exchange_polys = []
+            try:
+                for i in range(self._rank):
+                    self._exchange_polys.append(
+                        self._polynomial_ring(exchange_polys[i]))
+            except BaseException:
+                raise TypeError(("The supplied exchange polynomials are not"
+                                 " polynomials in the cluster variables over %s"
+                                 % (base_ring)))
+
+            self._coefficients = coefficients
+            self._check_seed()
+
+            # ensure mutation sequence / seed list initialises as empty lists
+            self._mutation_sequence = []
+
+            # begin with correct Laurent polynomials
+            self._laurent_polys = self._exchange_polys.copy()
+            self._compute_laurent()
+
+        else:
+            raise TypeError('Nonsupported input data. See documentation.')
+
+    # a private method to do some mathematical checks on our seed
+
+    def _check_seed(self):
+        r"""
+        Perform some mathematical checks on this seed. This includes
+        checking that each exchange polynomial does not depend on its
+        corresponding cluster variable, that each exchange polynomial is
+        irreducible, and finally that no cluster variable divides any exchange
+        polynomial.
+
+        TESTS::
+
+            sage: var('x1')
+            x1
+            sage: LpaSeed({x1: 1})
+            Traceback (most recent call last):
+            ...
+            ValueError: (LP2) fail: 1 is not irreducible over Integer Ring
+
+            sage: LpaSeed({x1: 0})
+            Traceback (most recent call last):
+            ...
+            ValueError: (LP2) fail: exchange polynomial is zero
+        """
+        # (LP1) check polynomials do not depend on their cluster variable
+
+        for i in range(0, self._rank):
+            currPoly = self._exchange_polys[i]
+            varList = currPoly.variables()
+            if self._names[i] in varList:
+                raise ValueError("(LP1) fail: %s depends on %s" %
+                                 (self._exchange_polys[i], self._names[i]))
+
+        # (LP2) check exchange polynomials are irreducible
+        # and not divisible by any cluster variable
+
+        for f in self._exchange_polys:
+            # check irreducibility
+            if f == 0:
+                raise ValueError("(LP2) fail: exchange polynomial is zero")
+            L = list(factor(f))
+            if (len(L) > 1) or f.is_unit():
+                raise ValueError("(LP2) fail: %s is not irreducible over %s"
+                                 % (f, self._base_ring))
+
+            # check not divisible by any variable
+            for var in self._names:
+                (q, r) = f.quo_rem(self._polynomial_ring(self._names[i]))
+                if r == 0:
```

Comment:
```suggestion
                if f % self._polynomial_ring(self._names[i]) == 0:
```
I am not sure if this works, but this would be the more natural/Pythonic thing IMO.